build_tools/cmake/iree_static_linker_test.cmake

```diff
@@ -52,8 +52,7 @@ function(iree_static_linker_test)
     return()
   endif()
 
-  if(NOT (IREE_TARGET_BACKEND_LLVM_CPU OR DEFINED IREE_HOST_BINARY_ROOT) OR
-     NOT IREE_HAL_DRIVER_LOCAL_SYNC)
+  if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR NOT IREE_HAL_DRIVER_LOCAL_SYNC)
```

Comment:
Interesting: https://github.com/iree-org/iree/actions/runs/3161388615/jobs/5147046310

`wasm-ld: error: unknown file type: tests/e2e/models/libiree_tests_e2e_models_simple_mul_llvm_cpu_static_c_test_lib.a(simple_mul_llvm_cpu_static_c_test.o)`

I might be able to look at that next week? Or we could exclude with the CMake `EMSCRIPTEN` variable too. Revert is fine for now (Friday afternoon)